internal/docs/openapi_loader.go

```diff
@@ -0,0 +1,266 @@
+		return
+	}
+
+	loader := openapi3.NewLoader()
+	loader.IsExternalRefsAllowed = true
+
+	doc, err := loader.LoadFromData(data)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Warning: Failed to parse OpenAPI spec: %v\n", err)
+		fmt.Fprintf(os.Stderr, "Field descriptions will not be enhanced with OpenAPI metadata.\n")
+		return
+	}
+
+	openAPIDoc = doc
+}
+
+// GetOpenAPIDescription retrieves an enhanced description for a field from the OpenAPI spec.
+// Priority order:
+// 1. Path parameters (components.parameters) - for fields ending in _id
+// 2. Header parameters (components.parameters) - for if_match
+// 3. Response headers (components.headers) - for etag
+// 4. Schema references (components.schemas) - for audit
+// 5. Schema properties (CreateXRequest, GetXResponse, etc.)
+// Converts snake_case field names to camelCase.
+// Returns empty string if schema or field not found.
+func GetOpenAPIDescription(resourceName, tfFieldName string) string {
+	if openAPIDoc == nil || openAPIDoc.Components == nil {
+		return ""
+	}
+
+	// Check for special fields first
+	switch tfFieldName {
+	case "if_match":
+		// Header parameter
+		if openAPIDoc.Components.Parameters != nil {
+			if paramRef, ok := openAPIDoc.Components.Parameters["If-Match"]; ok && paramRef.Value != nil {
+				if paramRef.Value.Description != "" {
+					return strings.TrimSpace(paramRef.Value.Description)
+				}
+			}
+		}
+
+	case "etag":
+		// Response header
+		if openAPIDoc.Components.Headers != nil {
+			if headerRef, ok := openAPIDoc.Components.Headers["ETag"]; ok && headerRef.Value != nil {
+				if headerRef.Value.Description != "" {
+					return strings.TrimSpace(headerRef.Value.Description)
+				}
+			}
+		}
+
+	case "audit":
+		// Schema reference
+		if openAPIDoc.Components.Schemas != nil {
+			if schemaRef, ok := openAPIDoc.Components.Schemas["CouchbaseAuditData"]; ok && schemaRef.Value != nil {
+				return "Couchbase audit data."
+			}
+		}
+
+	default:
+		// Check if this is a path parameter (e.g., organization_id, project_id)
+		if strings.HasSuffix(tfFieldName, "_id") && openAPIDoc.Components.Parameters != nil {
+			paramName := snakeToCapitalizedCamel(tfFieldName)
+			if paramRef, ok := openAPIDoc.Components.Parameters[paramName]; ok && paramRef.Value != nil {
+				if paramRef.Value.Description != "" {
+					return strings.TrimSpace(paramRef.Value.Description)
+				}
+			}
+		}
+	}
+
+	if openAPIDoc.Components.Schemas == nil {
+		return ""
+	}
+
+	// Convert snake_case to camelCase for schema property lookup
+	camelFieldName := snakeToCamel(tfFieldName)
+
+	// Capitalize resource name for schema patterns
+	capitalizedResource := capitalize(resourceName)
+
+	// Try common schema name patterns
+	schemaPatterns := []string{
+		"Create" + capitalizedResource + "Request",
+		"Get" + capitalizedResource + "Response",
+		"Update" + capitalizedResource + "Request",
+		capitalizedResource + "Request",
+		capitalizedResource + "Response",
+	}
```

Comment:
I think there are a few examples that this won't catch and it may be useful to pass the schema name to the file name? I am wondering if we could get the request/response schemas through [`openapi3.Operation`](https://pkg.go.dev/github.com/swaggest/openapi-go/openapi3#Operation) but that would require redesigning this a bit.